File: ProjectCS/UI/QuickItemPicker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SessionResultsCS.UI
{
    public class QuickItemPicker : BaseItemPicker
    {
        public QuickItemPicker(IEnumerable<string> items) 
            : base(items)
        {
            if (ItemCount > 9)
            {
                throw new ArgumentException("Maximum number of items is 9");
            }
        }

        protected override int GetInput()
        {
            Console.Write("Нажмите номер: ");
            while (true)
            {
                char input = Console.ReadKey(true).KeyChar;

                if (char.IsAsciiDigit(input))
                {
                    int value = (int)char.GetNumericValue(input);

                    if (value >= 1 && value <= ItemCount)
                    {
                        return value - 1;
                    }
                }
            }
        }
    }
}

File: Project/UI/QuickItemPicker.cpp
#include "UI/QuickItemPicker.hpp"

#include <iostream>
#include "conio.h"


QuickItemPicker::QuickItemPicker(const std::vector<std::string>& items) : BaseItemPicker(items) {
	if (GetItemCount() > 9) {
		throw std::invalid_argument("Maximum number of items is 9");
	}
}


int QuickItemPicker::GetInput() {
	std::cout << "Нажмите номер: ";

	while (true) {
		char c = _getch();
		int value = c - '1';

		if (value >= 0 && value < GetItemCount()) {
			return value;
		}
	}
}
